pkg/ddl/ddl.go

```diff
@@ -416,20 +449,51 @@ func (sv *schemaVersionManager) setSchemaVersion(job *model.Job, store kv.Storag
 }
 
 // lockSchemaVersion gets the lock to prevent the schema version from being updated.
-func (sv *schemaVersionManager) lockSchemaVersion(jobID int64) {
+func (sv *schemaVersionManager) lockSchemaVersion(jobID int64) error {
 	ownerID := sv.lockOwner.Load()
 	// There may exist one job update schema version many times in multiple-schema-change, so we do not lock here again
 	// if they are the same job.
 	if ownerID != jobID {
 		sv.schemaVersionMu.Lock()
 		sv.lockOwner.Store(jobID)
+		if sv.etcdClient != nil && variable.DDLVersion.Load() == model.TiDBDDLV2 {
+			se, err := concurrency.NewSession(sv.etcdClient)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			mu := concurrency.NewMutex(se, ddlSchemaVersionKeyLock)
+			if err := mu.Lock(sv.ctx); err != nil {
+				return errors.Trace(err)
+			}
+			sv.seMaps[jobID] = se
+			sv.lockMaps[jobID] = mu
+		}
 	}
+	return nil
 }
 
 // unlockSchemaVersion releases the lock.
 func (sv *schemaVersionManager) unlockSchemaVersion(jobID int64) {
 	ownerID := sv.lockOwner.Load()
 	if ownerID == jobID {
+		if sv.etcdClient != nil && variable.DDLVersion.Load() == model.TiDBDDLV2 {
+			se, ok := sv.seMaps[jobID]
+			if !ok {
+				return
+			}
+			mu, ok := sv.lockMaps[jobID]
+			if !ok {
+				return
+			}
+			delete(sv.seMaps, jobID)
+			delete(sv.lockMaps, jobID)
+			if err := mu.Unlock(sv.ctx); err != nil {
```

Comment:
Will this unlocking failure cause the lock to persist?

Reply:
yes, until the context done

Reply:
Shall we add retry first? It feels better than being stuck and unable to assign the schema version

Reply:
[b4a1d4b](https://github.com/pingcap/tidb/pull/49687/commits/b4a1d4b524d3fde5653de7fc1efd088dc523ffda) add retry